### ISSUE TITLE
Cannot install WCF Connected Services without Microsoft Connected Services

### ISSUE THREAD

**Issue by VeXHarbinger** [Microsoft/ConnectedServices#4]
Howdy All,

I just cannot figure this out.  My IDE offers an **Add Connected Service** action with a few service options, just not the WCF one.
Every time I try to follow the steps to add the [WCF Service option](https://blogs.msdn.microsoft.com/webdev/2016/06/26/wcf-connected-service-for-net-core-1-0-0-and-asp-net-core-1-0-0-is-now-available/) I receive the following error;

```
Install Error : Microsoft.VisualStudio.ExtensionManager.MissingReferencesException: This extension cannot be installed because the following references are missing:
-ConnectedServices.B3144AF8-5B8F-491C-9ACA-996350E8FE95 (Microsoft Connected Services)
```

I do have both of the requirements installed;

- Visual Studio 2015 Update 3
- .Net Core and ASP.NET Core 1.0 SDK

So I've gone around and around reinstalling these components, running a repair on my IDE but none seem to replace the missing Connected Service component it's looking for.
Any steps/ideas on how to resolve this or if there is a component that causes this error?  I've need no results on Google and I'm at a loss as to what else I could do.

**Comment by VeXHarbinger** [comment]
Installer Log
[VSIXInstaller_7607df9a-fba4-4c85-815e-1d9789c38bff.log.txt](https://github.com/Microsoft/ConnectedServices/files/790628/VSIXInstaller_7607df9a-fba4-4c85-815e-1d9789c38bff.log.txt)

**Comment by VeXHarbinger** [comment]
Anyone have any ideas?  I'm still stuck with this issue

**Comment by viperium** [comment]
I'm experiencing the same problem.
Both for WCF Connected Service and [OData Connected Service](https://marketplace.visualstudio.com/items?itemName=laylaliu.ODataConnectedService)

How can I add a reference to Microsoft Connected Services? :disappointed:

**Comment by sjc01234** [comment]
I was also having this problem: got the extension installed (and restarted VS2015) but could not see WCF as an option to add a connected service.  I had already started a .net core console app project before installing the extension and was attempting to add the connected service in this existing project.

So, I started two new projects:

- .net core web app (blank/empty) -> I was able to see WCF as an option.

- .net core console app -> I was able to see WCF as an option

So, not sure if this is actually a solution, but for me it appears that you need to start a new project after getting the extension installed.

**Comment by VeXHarbinger** [comment]
Nope, i'm 100% blocked on this one.  Had to take development to a new machine

**Comment by dmitry-pavlov** [comment]
The issue is still here - I have just got this while updating my VS2017 extension to support VS2019 preview. 

I have reported this bug here - **please vote for fixing it**: [VS2019 Preview: cannot install connected service extension updated from VS2017](https://developercommunity.visualstudio.com/content/problem/468751/vs2019-preview-cannot-install-connected-service-ex.html)

**Comment by dmitry-pavlov** [comment]
According to Angelos Petropoulos [MSFT] · [17 hours ago](https://developercommunity.visualstudio.com/content/problem/468751/vs2019-preview-cannot-install-connected-service-ex.html?childToView=483882#comment-483882)
> Microsoft Connected Services have been deprecated in VS 2019. Please use standard Visual Studio extensibility points instead: https://docs.microsoft.com/en-us/visualstudio/extensibility/

**Comment by AndrewBrianHall** [comment]
Saying that connected services is deprecated is not quite the right messaging.  The team had been working on a set of new APIs that would provide richer functionality that the existing APIs and had take some proactive steps to try to help people not start new things on the old APIs.  However, it was a mistake that existing extensions can't be installed and this will be fixed in the first update to Visual Studio 2019 (version 16.1)

**Comment by unchase** [comment]
Thank you, we will wait for the update.

**Comment by CoolDadTx** [comment]
If there is a new set of APIs coming but we need to support connected services now what are the options? 16.1 will fix the issue that you cannot install a CS extension but what about even trying to build one? Trying to use the Microsoft.VisualStudio.ConnectedServices 16.0 NuGet package will still fail because the Microsoft.Developer.IdentityService.Client package isn't available in NuGet. We could go back to 15.3.47 which seems to be the last version of the Package that doesn't require it but how will that impact trying to build an extension for VS 2019?
